roles/edpm_network_config/tasks/main.yml

```diff
@@ -23,23 +23,8 @@
     - edpm_network_config_tool == 'nmstate'
     - not edpm_network_config_tool_nmstate_override|bool
 
-- name: Install OVS NetworkManager plugin [nmstate and nmstate provider]
-  when: edpm_network_config_tool == 'nmstate' or edpm_network_config_nmstate | bool
-  become: true
-  block:
-    - name: Install OVS NetworkManager plugin [nmstate]
-      ansible.builtin.dnf:
-        name: "{{ edpm_network_config_systemrole_nmstate_dependencies }}"
-        state: present
-      register: nm_ovs_status
-      until: nm_ovs_status is succeeded
-      retries: "{{ edpm_network_config_download_retries }}"
-      delay: "{{ edpm_network_config_download_delay }}"
-    - name: Restart NetworkManager after plugin installation [nmstate]
-      ansible.builtin.systemd:
-        name: NetworkManager
-        state: restarted
-      when: nm_ovs_status.changed  # noqa: no-handler
+- name: Install required packages
+  ansible.builtin.import_tasks: install.yml
 
 - name: Import DNS NetworkManager configs tasks
   ansible.builtin.import_tasks: dns_nm_configs.yml
```

Comment:
This would still not work after minor updates when edpm_network_config_nmstate default is changed to true (earlier it would be false) which would remove dns=none and you want to re-run `configure_network` service (probably part of scale-out) 

This is what will happen

- Reloading NetworkManager would wipe out /etc/resolv.conf as cloud-init would have configured network with sysconfig renderer and existing nodes will permanently lose entries in resolv.conf.
- There would be no os-net-config run on existing nodes as the os-net-config has run once to reconfigure resolv.conf.

I would suggest we add a check to not touch NetworkManager.conf for existing nodes where network has already been configured.

Please test all scenarios including minor updates.

Reply:
@rabi only Network Manager reload causing this issue for existing nodes where network already configured.
Only Network Manager reload triggered when Network Manager conf  changes done first time either removed or set dns line.

Reply:
@rabi before default was nmstate provider mean, ifcfg to nmstate provider change is not supported in minor update scenario, only same provider to be used in minor update either ifcfg or nmstate.
We are working on migration (ifcfg to nmstate provider) and in progress.

Reply:
> @rabi before default was nmstate provider mean, ifcfg to nmstate provider change is not supported in minor update scenario, only same provider to be used in minor update either ifcfg or nmstate. We are working on migration (ifcfg to nmstate provider) and in progress.

@karthiksundaravel @vcandapp ^

Reply:
How do you restrict it when you say unsupported? Earlier default was ifcfg and new default is nmstate. Without any change in CRs, update to FR3 will change the default and it would break things.